src/collector/tweak_score_top_collector.rs

```diff
@@ -121,9 +121,112 @@ where
+    use crate::schema::*;
+    use crate::{DocId, Index, Score, SegmentReader};
+    use std::collections::HashSet;
+
+    #[test]
+    fn tweak_score_with_facets() {
+        let mut schema_builder = Schema::builder();
+
+        let title = schema_builder.add_text_field("title", STORED);
+        let ingredient = schema_builder.add_facet_field("ingredient");
+
+        let schema = schema_builder.build();
+        let index = Index::create_in_ram(schema.clone());
+
+        let mut index_writer = index.writer(30_000_000).unwrap();
+
+        index_writer.add_document(doc!(
+            title => "Fried egg",
+            ingredient => Facet::from("/ingredient/egg"),
+            ingredient => Facet::from("/ingredient/oil"),
+        ));
+        index_writer.add_document(doc!(
+            title => "Scrambled egg",
+            ingredient => Facet::from("/ingredient/egg"),
+            ingredient => Facet::from("/ingredient/butter"),
+            ingredient => Facet::from("/ingredient/milk"),
+            ingredient => Facet::from("/ingredient/salt"),
+        ));
+        index_writer.add_document(doc!(
+            title => "Egg rolls",
+            ingredient => Facet::from("/ingredient/egg"),
+            ingredient => Facet::from("/ingredient/garlic"),
+            ingredient => Facet::from("/ingredient/salt"),
+            ingredient => Facet::from("/ingredient/oil"),
+            ingredient => Facet::from("/ingredient/tortilla-wrap"),
+            ingredient => Facet::from("/ingredient/mushroom"),
+        ));
+        index_writer.commit().unwrap();
+
+        let reader = index.reader().unwrap();
+        let searcher = reader.searcher();
+        {
+            let facets = vec![
+                Facet::from("/ingredient/egg"),
+                Facet::from("/ingredient/oil"),
+                Facet::from("/ingredient/garlic"),
+                Facet::from("/ingredient/mushroom"),
+            ];
+            let query = BooleanQuery::new_multiterms_query(
+                facets
+                    .iter()
+                    .map(|key| Term::from_facet(ingredient, &key))
+                    .collect(),
+            );
+            let top_docs_by_custom_score =
+                TopDocs::with_limit(2).tweak_score(move |segment_reader: &SegmentReader| {
+                    let mut ingredient_reader = segment_reader.facet_reader(ingredient).unwrap();
+                    let facet_dict = ingredient_reader.facet_dict();
+
+                    let query_ords: HashSet<u64> = facets
+                        .iter()
+                        .filter_map(|key| facet_dict.term_ord(key.encoded_str()))
+                        .collect();
+
+                    let mut facet_ords_buffer: Vec<u64> = Vec::with_capacity(20);
+
+                    move |doc: DocId, original_score: Score| {
+                        ingredient_reader.facet_ords(doc, &mut facet_ords_buffer);
+                        let missing_ingredients = facet_ords_buffer
```

Comment:
That seems very suboptimal... What about just iter().filter().count()?

Reply:
Good point, this was here for debugging but when I was figuring things out but doesn't need to be any more.